prow/capo-cluster/kubeadmcontrolplane.yaml

```diff
@@ -15,16 +15,22 @@ spec:
       nodeRegistration:
         kubeletExtraArgs:
           cloud-provider: external
+          provider-id: "openstack:///'{{ instance_id }}'"
```

Comment:
humm? how was it working absent provider-id?

Reply:
When this is not specified, the cloud controller compares the names of the OpenStackMachines or Node with the names of the servers in the OpenStack API and sets the provider ID based on that.

We saw some issues in CAPO when names contains special characters, because OpenStack modifies the name automatically for the server but this is not done for Nodes and other resources. That way you can get mismatch and then the cloud controller is unable to set the provider ID. This is why we have now added it to avoid these issues.